### PR TITLE
Decouple RGB Matrix from RGB Light

### PR DESCRIPTION
**Pull request by drashna**
## Description

Right now, the RGB Matrix feature works by taking control of a number of the RGB Light functions, when it's disabled.  

This is fine if you want to run one or the other. But if you're a madman like me, and want to run both, there is no choice.  You are only able to control the RGB Light feature from keycodes.  

This changes the functionality so that the `rgb_matrix_*` functions are called explicitly, so that there is no need to to remap the functions. 

However, this would mean that if you have both features enabled at the same time, it would control both sets of features.  This is also less than idea.  So a `RGB*_DISABLE_KEYCODES` setting has been added so that you can selective which features are controlled by the keycodes (rather than introducing a new set of keycodes, and remapping)
However, I think that adding RGB_MATRIX keycodes and remapping if the RGBLIGHT feature is disabled would be best. Though, I'm not sure of what keycode prefix should be used. 
 
## Types of Changes

- [X] Core
- [ ] Bugfix
- [ ] New feature
- [X] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [X] Documentation

## Issues Fixed or Closed by This PR

* Drashna's sanity (insanity?), and his Planck Light

## Checklist

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).
   - I have tested that this works on a board with just RGB Light enabled, just RGB Matrix enabled, and both enabled. As well as with neither enabled.  All compile and appear to work.

### CODE REVIEW

**Comment by drashna** [comment]
Posting as draft, because there are a few parts that I feel need feedback before it should be considered for merging.

File: docs/feature_rgblight.md [hunk 1]
```diff
@@ -64,6 +64,9 @@ Changing the **Value** sets the overall brightness.
 |`RGB_MODE_GRADIENT`|`RGB_M_G` |Static gradient animation mode                                      |
 |`RGB_MODE_RGBTEST` |`RGB_M_T` |Red, Green, Blue test animation mode                                |
 
+!> By default, if you have both the [RGB Light](feature_rgblight.md) and the RGB Matrix feature enabled, these keycodes will work for both features, at the same time. You can disable the keycode functionality by defining the `*_DISABLE_KEYCODES` option for the specific feature.
```
Comment (noroadsleft) [inline docs/feature_rgblight.md]: Linking the wrong text to the wrong file here.

```suggestion
!> By default, if you have both the RGB Light and the [RGB Matrix](feature_rgb_matrix.md) feature enabled, these keycodes will work for both features, at the same time. You can disable the keycode functionality by defining the `*_DISABLE_KEYCODES` option for the specific feature.
```

**Comment by drashna** [comment]
Since the last thing that this really does is add new handling for the keycodes. Holding off on this until #7512 is merged, since it centralizes the keycodes, and makes things easier to manage in the long run.

**Comment by drashna** [comment]
Closing.  Will reopen in some form when #7508 gets merged.

**Pull request closed without merging**